Dockerfile

```diff
@@ -1,12 +1,16 @@
-FROM node:latest
-LABEL maintainer="https://github.com/4x0v7"
+FROM node:alpine
+
+# Use the Open Container Image Annotation Spec (https://github.com/opencontainers/image-spec/blob/master/annotations.md)
+LABEL org.opencontainers.image.title="markdown-link-check"
+LABEL org.opencontainers.image.description="checks that all hyperlinks in a markdown text to determine if they are alive or dead"
+LABEL org.opencontainers.image.documentation="https://github.com/tcort/markdown-link-check/blob/master/README.md"
+LABEL org.opencontainers.image.source="https://github.com/tcort/markdown-link-check"
 
 # Install app dependencies
-COPY package.json /src/package.json
+COPY package*.json /src/
 WORKDIR /src
 RUN set -ex; \
-    npm install \
-    && npm ls
+    npm install
```

Comment:
could we use `npm ci` here instead to make sure what's installed is what's in the package-lock.json ?

Reply:
I don't know the ins and outs of NPM, happy to change it if you think it's best. It might be good to change build methodology in a separate PR though. Honestly, I had to install NPM to work with markdown-link-check. 😂 Not sure how I avoided it so long.

Not directly related, I don't think the `npm test` belongs in the build step but it doesn't cause any issues other than longer build times.

I'll try to put together my thoughts on  testing and start a conversation when I find some downtime today.

Reply:
I agree the tests should be a step in the release process that will then build the image once the test succeeds. Here there's nothing that even prevents the image to build if tests fails so it's a bit useless.

For the `npm ci` vs `npm install` it's related to the image a lot. Let me explain:

Using a docker image first goal is to have a reproductible environment.
All known and tested versions, no surprise.

If you use `npm install` it will look at the package.json file and if any dependency has an available update for instance [because you used `^1.2.3` as the version number](https://docs.npmjs.com/files/package.json#dependencies), it means that we allow to install the latest version >=1.2.3 but <2.0.0.

In the case when the dependency is updated it means that the Docker image can include any version between 1.2.3 and 2.0.0 if we use `npm install`.

If we use `npm ci`, it will install the exact version indicated in the package-lock.json. As we commit this file and tested before pushing that it works (CI test before building image have to be here) is ensures that the version in the Docker image is exactly the one we know and strengthen the Docker paradigm of an image providing a well known and stable environment (aka "It will prevent crazy bugs").

Reply:
I will take the responsibility to move to `mpn ci` so that we can merge this one.